File: haskell/08/pizza.hs
import System.IO

putStrF s = do
    putStr s
    hFlush stdout

ask q f = do
    putStrF (q ++ " ")
    f

getInteger = do
    hSetEcho stdin False
    hSetBuffering stdin NoBuffering
    n <- getIntegerRec ""
    hSetEcho stdin True
    return n

getIntegerRec s = do
    c <- getChar
    if c == '\n' || (c >= '0' && c <= '9')
    then do
        putChar c
        hFlush stdout
        if c == '\n' then return s else getIntegerRec (s ++ [c])
    else getIntegerRec s

promptInt q = do
    nStr <- ask q getInteger
    return (read nStr :: Integer)


data Piece = Piece Integer

piece 1 = "piece"
piece n = "pieces"

main = do
    nPeople <- promptInt "How many people?"
    nPizza <- promptInt "How many pizzas do you have?"
    nPiece <- promptInt "How many pieces are in a pizza?"
    let totalPiece = nPiece * nPizza
    let piecePerPerson = totalPiece `div` nPeople
    let leftOver = totalPiece - (nPeople * piecePerPerson)
    putStrLn $ "Each person gets " ++ (show piecePerPerson) ++ 
        " " ++ (piece piecePerPerson) ++ " of pizza."
    putStrLn $ "There are " ++ (show leftOver) ++ 
        " leftover " ++ (piece leftOver) ++ "."

File: python/08/pizza.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

num_people = int(input("How many people? "))
num_pizza = int(input("How many pizzas do you have? "))
num_piece = int(input("\nHow many pieces are in a pizza? "))

num_all_piece = num_piece * num_pizza
piece_per_person = num_all_piece // num_people
num_leftover = num_all_piece - piece_per_person * num_people

person_str = "person" if num_people == 1 else "people"
pizza_str = "pizza" if num_pizza == 1 else "pizzas"
per_piece_str = "piece" if piece_per_person == 1 else "pieces"

leftover_be_str = "is" if piece_per_person == 1 else "are"
leftover_piece_str = "piece" if piece_per_person == 1 else "pieces"

print(f"{num_people} {person_str} with {num_pizza} {pizza_str}")
print(f"Each person gets {piece_per_person} {per_piece_str} of pizza")
print(f"There {leftover_be_str} {num_leftover} leftover {leftover_piece_str}")
